hack4impact-utk/community-coalition-against-human-trafficking
# CCAHT‐45 Added text and number attribute support.

- Dynamically created TextField components for text and number attributes based on components of `splitAttrs.text` and `splitAttrs.number`.
- Created a function (`updateTextFieldAttributes`) that updates text and number attributes, called in the `onChange` prop of the TextField components.

To test:
Add this on the check in page (edited from CCAHT-39):
```ts
import { Attribute } from 'utils/types'
const attrs: Attribute[] = [
  {
    _id: '1',
    name: 'Color',
    possibleValues: ['Red', 'Blue', 'Green'],
    color: '#5e9cff',
  },
  {
    _id: '2',
    name: 'Size',
    possibleValues: ['Small', 'Medium', 'Large'],
    color: '#93fab7',
  },
  {
    _id: '3',
    name: 'Text Attr 1',
    possibleValues: 'text',
    color: '#00ff00',
  },
  {
    _id: '4',
    name: 'Text Attr 2',
    possibleValues: 'text',
    color: '#00ff00',
  },
  {
    _id: '5',
    name: 'Num Attr 1',
    possibleValues: 'number',
    color: '#00ff00',
  },
  {
    _id: '6',
    name: 'Num Attr 2',
    possibleValues: 'number',
    color: '#00ff00',
  },
]
```
And pass in attrs as the attribute prop for `<CheckInCheckOutForm ... />`.
